pandas/tests/indexing/test_partial.py

```diff
@@ -637,3 +637,66 @@ def test_partial_set_empty_frame_empty_consistencies(self):
         df.loc[0, 'x'] = 1
         expected = DataFrame(dict(x=[1], y=[np.nan]))
         tm.assert_frame_equal(df, expected, check_dtype=False)
+
+    def test_access_timezoned_datetimeindex_with_timezoned_label_utc(self):
+
+        # GH 6785
+        # timezone was ignored when string was provided as a label
+
+        first_january = pd.date_range('2016-01-01T00:00', '2016-01-01T23:59',
+                                      freq='T', tz="UTC")
+        df = pd.DataFrame(index=first_january, data=np.arange(len(
+                          first_january)))
+
+        result = df[
+            "2016-01-01T00:00-02:00":"2016-01-01T02:03"
+        ]
+
+        expected = df[
+            pd.Timestamp("2016-01-01T00:00-02:00"):
+            pd.Timestamp("2016-01-01T02:03")
+        ]
+
+        tm.assert_frame_equal(result, expected)
```

Comment:
Can you assert both results (with strings or with Timestamps) with an independelty constructed one? (eg `df.iloc[...]`)

Reply:
I am not sure what you mean with `df.iloc[...]` but generally speaking yes

Reply:
I mean to created 'expected' with something like df.iloc[120:124] (but then with the cirrect numbers)